### ISSUE TITLE
[Feature Request] Overclocking the DS CPU

### ISSUE THREAD

**Issue by MojoJojoDojo** [Arisotura/melonDS#464]
The DS CPU Is clocked to 67 mhz while the DSi CPU is clocked to 133 mhz.
It would be nice if you could overclock/downclock the emulated CPU to the DSi clock speed or if its
possible than even higher then 133 mhz (like Dolphin could do) for both DSi-mode games and regular DS-mode games.
Basically forcing a higher clock speed for all games.

**Comment by Arisotura** [comment]
indeed, we might abuse the mechanism to set the ARM9 clock to any multiple of the 33MHz system clock.

but first I need to implement DSi clock switching without making things shit themselves.

**Comment by MojoJojoDojo** [comment]
Seeing as DSi work is complete is it time for the feature to be implemented?
I am mostly talking about implementing this to normal DS games that cant benefit from the DSi CPU clocks.

**Comment by RafaelTrepaUnCarballo** [comment]
I'm interested in this feature.

In a real DSi/3DS is possible to overclock DS games by forcing them to run in DSi mode with nds-bootstrap.
That rises the CPU clocks (67->133MHz) and the audio frequency (32->48kHz), reducing slowdowns and loadtimes in some games (like gen4 pokemon), although it's problematic with other games.
